internal/server/acl/utils.go

```diff
@@ -0,0 +1,72 @@
+package acl
+
+import (
+	"slices"
+	"sort"
+	"strings"
+
+	"github.com/openmined/syftbox/internal/aclspec"
+)
+
+func calculateGlobSpecificity(glob string) int {
+	// early return for the most specific glob patterns
+	switch glob {
+	case "**":
+		return -100
+	case "**/*":
+		return -99
+	}
+
+	// 2L + 10D - wildcard penalty
+	// Use forward slash for glob patterns
+	score := len(glob)*2 + strings.Count(glob, ACLPathSep)*10
+
+	if hasTemplatePattern(glob) {
+		score += 50
+	}
+
+	// penalize base score for substr wildcards
+	for i, c := range glob {
+		switch c {
+		case '*':
+			if i == 0 {
+				score -= 20 // Leading wildcards are very unspecific
+			} else {
+				score -= 10 // Other wildcards are less penalized
+			}
+		case '?', '!', '[', '{':
+			score -= 2 // Non * wildcards get smaller penalty
+		}
+	}
+
+	return score
+}
+
+func sortRulesBySpecificity(rules []*aclspec.Rule) []*aclspec.Rule {
```

Comment:
We probably need more tests for this in case of weird edge cases, sort order and tie breaks etc.

Reply:
Added tests for this [a5714d1](https://github.com/OpenMined/syftbox/pull/62/commits/a5714d1e444c2e7a45a89c1cbfff402a7215f4ca)